src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -356,6 +358,56 @@ else if (!period.isEmpty()) {
             .map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement);
     }
 
+    /**
+     * Whether the {@code {@summary}} tag is defined first in the javadoc.
+     *
+     * @param inlineSummaryTag node of type {@link JavadocTokenTypes#JAVADOC_INLINE_TAG}
+     * @return {@code true} if the {@code {@summary}} tag is defined first in the javadoc
+     */
+    private static boolean isDefinedFirst(DetailNode inlineSummaryTag) {
+        boolean isDefinedFirst = true;
+        DetailNode previousSibling = JavadocUtil.getPreviousSibling(inlineSummaryTag);
+        while (previousSibling != null && isDefinedFirst) {
+            final int siblingType = previousSibling.getType();
+            if (siblingType == JavadocTokenTypes.TEXT) {
+                if (!previousSibling.getText().isBlank()) {
+                    isDefinedFirst = false;
+                }
+            }
+            else if (siblingType == JavadocTokenTypes.HTML_ELEMENT
+                && isTextPresentInsideHtmlTag(previousSibling)) {
+                isDefinedFirst = false;
+            }
+            previousSibling = JavadocUtil.getPreviousSibling(previousSibling);
+        }
+        return isDefinedFirst;
+    }
+
+    /**
+     * Whether some text is present inside the HTML element or tag.
+     *
+     * @param node DetailNode of type {@link JavadocTokenTypes#HTML_TAG}
+     *             or {@link JavadocTokenTypes#HTML_ELEMENT}
+     * @return {@code true} if some text is present inside the HTML element or tag
+     */
+    public static boolean isTextPresentInsideHtmlTag(DetailNode node) {
+        DetailNode nestedChild = JavadocUtil.getFirstChild(node);
+        if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+            nestedChild = JavadocUtil.getFirstChild(nestedChild);
+        }
+        boolean isTextPresentInsideHtmlTag = false;
+        while (nestedChild != null && !isTextPresentInsideHtmlTag) {
+            final int nestedChildType = nestedChild.getType();
+            isTextPresentInsideHtmlTag = nestedChildType == JavadocTokenTypes.TEXT
+                && !nestedChild.getText().isBlank()
+                || (nestedChildType == JavadocTokenTypes.HTML_TAG
+                || nestedChildType == JavadocTokenTypes.HTML_ELEMENT)
```

Comment:
please add indentation. Without it this line looks like it belongs to main condition chain.
As a side note - may be switch will look better than 5 conditions? e.g.
```
switch(nestedChild.getType()) {
   case JavadocTokenTypes.TEXT:
       isTextPresentInsideHtmlTag = !nestedChild.getText().isBlank();
       break;
   case JavadocTokenTypes.HTML_TAG:
   case JavadocTokenTypes.HTML_ELEMENT:
        isTextPresentInsideHtmlTag = isTextPresentInsideHtmlTag(nestedChild);
        break;
   default:
      isTextPresentInsideHtmlTag =false;
      break;
}
```
it may be a bit more verbose but definitely more readable, as for me

Reply:
same applies for method above, can be like
```
swithc(previousSibling.getType()) {
            case JavadocTokenTypes.TEXT: {
                isDefinedFirst previousSibling.getText().isBlank(); break;
                }
            }
            case JavadocTokenTypes.HTML_ELEMENT:
                isDefinedFirst = !isTextPresentInsideHtmlTag(previousSibling); break;
            }
            default:  isDefinedFirst = true;
```

what you think?

Reply:
Good suggestion, implemented.